CMakeLists.txt

```diff
@@ -271,6 +235,8 @@ add_llvm_library(${TARGET_NAME} SHARED
     ${ADDITIONAL_LIBS}
     ${CMAKE_DL_LIBS})
 
+llvm_config(${TARGET_NAME} all)
```

Comment:
this should probably be llvm_config(${TARGET_NAME} USE_SHARED all) if I'm reading https://github.com/llvm-mirror/llvm/blob/master/cmake/modules/LLVM-Config.cmake#L67 correctly

Reply:
Please take a look at https://github.com/intel/opencl-clang/pull/53

Reply:
@tripzero : you might need USE_SHARED if you build against LLVM built into dynamic library, but I built LLVM into range of static libraries (well, this was a default) and if I follow your suggestion I get:
```
[ 92%] Linking CXX shared library libcommon_clang.so
/opt/rh/devtoolset-7/root/usr/libexec/gcc/x86_64-redhat-linux/7/ld: cannot find -lLLVM
```

Hence, some condition might be needed to differentiate between 2 kind of LLVM builds: dynamic and static. I originally missed this attribute and thought hat llvm_config will take care about differentiation on its own.

I see that @AlexeySotkin provides slightly different way to handle the same in #53. Let's take a look on it, it might be a better option if it provides internal differentiation between dynamic or static LLVM builds